### PR TITLE
Follow-up fixes from #143

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:
- Removes the Cluster.Spec.DefaultZone
- Makes sure to set the NetworkID as part of the instance's Network Interface definition
- Updates CAPI to latest master
- Uses the record package to record events

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: cloud/services/compute/instances.go [hunk 1]
```diff
@@ -68,11 +68,13 @@ func (s *Service) CreateInstance(scope *scope.MachineScope) (*compute.Instance,
...
 	input := &compute.Instance{
-		Name:              scope.Name(),
-		Zone:              scope.Zone(),
-		MachineType:       fmt.Sprintf("zones/%s/machineTypes/%s", scope.Zone(), scope.GCPMachine.Spec.InstanceType),
-		CanIpForward:      true,
-		NetworkInterfaces: []*compute.NetworkInterface{{}},
+		Name:         scope.Name(),
+		Zone:         scope.Zone(),
+		MachineType:  fmt.Sprintf("zones/%s/machineTypes/%s", scope.Zone(), scope.GCPMachine.Spec.InstanceType),
+		CanIpForward: true,
```
Comment (justinsb) [inline cloud/services/compute/instances.go]: BTW if we're using Calico I don't think we need CanIpForward...

**Comment by justinsb** [comment]
Thanks for the follow-up @vincepri !

/approve
/lgtm

File: main.go [hunk 2]
```diff
@@ -133,18 +134,19 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Initialize event recorder.
+	record.InitFromRecorder(mgr.GetEventRecorderFor("aws-controller"))
```
Comment (tahsinrahman) [inline main.go]: ```gcp-controller```

Comment (vincepri) [reply main.go]: whoops! Are you cool with fixing it as part of the other PR you have open?

**Pull request merged**